.gitignore

```diff
@@ -1,3 +1,4 @@
 .deps
 ponycc/ast/gen/gen
 ponycc/test/test
+ponycc/**/*.dSYM
```

Comment:
I'd have a preference for keeping this rule out of the `.gitignore` (and the related `Makefile`), since it's not directly related to anything that is specific to this project - that is, I would usually ask users to place this in their personal `.gitignore` (at the user level of their development machine).

That said, if you press me to include it here, I suppose I can allow it.

Reply:
Yeah, sounds good. I didn't even realize it was a mac-specific thing until I was writing the PR.
Will just add it to my global gitignore

Reply:
No, that is the right call IMO.. I didn't even realize it was mac-specific until writing this PR.

Will remove and add it to my own global config!

Reply:
Will also remove from the makefile for now as well